### PR TITLE
Remove PackageDescription API v3

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
@swift-ci test with toolchain

File: Tests/PackageLoadingTests/ManifestTests.swift [hunk 1]
```diff
@@ -1,289 +0,0 @@
...
-            let foo = manifest.targets[0]
-            XCTAssertEqual(foo.name, "Foo")
-            XCTAssertEqual(foo.dependencies, [.target(name: "Bar")])
-            XCTAssertEqual(manifest.dependencies, [])
-        }
-    }
-    
-    /// Check that we load the manifest appropriate for the current version, if
-    /// version specific customization is used.
-    func testVersionSpecificLoading() throws {
```
Comment (aciidb0mb3r) [inline Tests/PackageLoadingTests/ManifestTests.swift]: should port this test to 4.2

File: Tests/PackageLoadingTests/ManifestTests.swift [hunk 2]
```diff
@@ -1,289 +0,0 @@
...
-
-        stream = BufferedOutputByteStream()
-        stream <<< "import PackageDescription" <<< "\n"
-        stream <<< "let package = Package(" <<< "\n"
-        stream <<< "   name: \"Foo\")" <<< "\n"
-        manifest = try loadManifest(stream.bytes)
-        XCTAssertEqual(manifest.swiftLanguageVersions, nil)
-    }
-
-    func testRuntimeManifestErrors() throws {
```
Comment (aciidb0mb3r) [inline Tests/PackageLoadingTests/ManifestTests.swift]: port this too (if not already ported)

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci test with toolchain

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by huonw** [comment]
Unfortunately it looks like this broke CI https://ci.swift.org/job/oss-swift-package-linux-ubuntu-16_04/1899/, e.g.:

```
02:00:07 FAIL: swift-package-tests :: test-complex-xctest-package/test-xctest-package.txt (11 of 22)
02:00:07 ******************** TEST 'swift-package-tests :: test-complex-xctest-package/test-xctest-package.txt' FAILED ********************
02:00:07 Script:
02:00:07 --
02:00:07 rm -rf /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.dir
02:00:07 mkdir -p /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.dir
02:00:07 cp -r /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/swift-integration-tests/test-complex-xctest-package/SwiftCMixed /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.dir/
02:00:07 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/usr/bin/swift test --package-path /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.dir/SwiftCMixed -v 2>&1 | tee /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.build-log
02:00:07 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/llvm-linux-x86_64/bin/FileCheck --check-prefix CHECK-BUILD-LOG --input-file /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.build-log /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/swift-integration-tests/test-complex-xctest-package/test-xctest-package.txt
02:00:07 --
02:00:07 Exit Code: 1
02:00:07 
02:00:07 Command Output (stdout):
02:00:07 --
02:00:07 $ "rm" "-rf" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.dir"
02:00:07 $ "mkdir" "-p" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.dir"
02:00:07 $ "cp" "-r" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/swift-integration-tests/test-complex-xctest-package/SwiftCMixed" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.dir/"
02:00:07 $ "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/usr/bin/swift" "test" "--package-path" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.dir/SwiftCMixed" "-v"
02:00:07 note: command had no output on stdout or stderr
02:00:07 error: command failed with exit status: 1
02:00:07 $ "tee" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.build-log"
02:00:07 # command output:
02:00:07 which clang
02:00:07 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/usr/bin/swiftc --driver-mode=swift -L /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/usr/lib/swift/pm/4_2 -lPackageDescription -suppress-warnings -swift-version 4.2 -I /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/usr/lib/swift/pm/4_2 -sdk / /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.dir/SwiftCMixed/Package.swift -fileno 7
02:00:07 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.dir/SwiftCMixed: error: manifest parse error(s):
02:00:07 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.dir/SwiftCMixed/Package.swift:6:9: error: cannot invoke initializer for type 'Target' with an argument list of type '(name: String, dependencies: [String])'
02:00:07         Target(name: "swifty", dependencies: ["see"]),
02:00:07         ^
02:00:07 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-complex-xctest-package/Output/test-xctest-package.txt.tmp.dir/SwiftCMixed/Package.swift:6:9: note: expected an argument list of type '(name: String, dependencies: [Target.Dependency], path: String?, exclude: [String], sources: [String]?, publicHeadersPath: String?, type: Target.TargetType, pkgConfig: String?, providers: [SystemPackageProvider]?)'
02:00:07         Target(name: "swifty", dependencies: ["see"]),
02:00:07         ^
02:00:07 
02:00:07 
02:00:07 --
```

There's some other failures that aren't so obviously caused by this, but could be too:

```
02:00:20 FAIL: swift-package-tests :: test-foundation-package/test-foundation-package.txt (15 of 22)
02:00:20 ******************** TEST 'swift-package-tests :: test-foundation-package/test-foundation-package.txt' FAILED ********************
02:00:20 Script:
02:00:20 --
02:00:20 rm -rf /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.dir
02:00:20 mkdir -p /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.dir/tool
02:00:20 cp /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/swift-integration-tests/test-foundation-package/test-foundation-package.txt /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.dir/tool/Package.swift
02:00:20 cp /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/swift-integration-tests/test-foundation-package/main.swift /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.dir/tool/main.swift
02:00:20 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/usr/bin/swift build --package-path /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.dir/tool -v 2>&1 | tee /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.build-log
02:00:20 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/llvm-linux-x86_64/bin/FileCheck --check-prefix CHECK-BUILD-LOG --input-file /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.build-log /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/swift-integration-tests/test-foundation-package/test-foundation-package.txt
02:00:20 test -x /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.dir/tool/.build/debug/tool
02:00:20 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.dir/tool/.build/debug/tool > /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.out
02:00:20 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/llvm-linux-x86_64/bin/FileCheck --check-prefix CHECK-TOOL-OUTPUT --input-file /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.out /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/swift-integration-tests/test-foundation-package/test-foundation-package.txt
02:00:20 --
02:00:20 Exit Code: 1
02:00:20 
02:00:20 Command Output (stdout):
02:00:20 --
02:00:20 $ "rm" "-rf" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.dir"
02:00:20 $ "mkdir" "-p" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.dir/tool"
02:00:20 $ "cp" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/swift-integration-tests/test-foundation-package/test-foundation-package.txt" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.dir/tool/Package.swift"
02:00:20 $ "cp" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/swift-integration-tests/test-foundation-package/main.swift" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.dir/tool/main.swift"
02:00:20 $ "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/usr/bin/swift" "build" "--package-path" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.dir/tool" "-v"
02:00:20 $ "tee" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.build-log"
02:00:20 # command output:
02:00:20 lsb_release -r
02:00:20 which clang
02:00:20 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/usr/bin/swiftc --driver-mode=swift -L /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/usr/lib/swift/pm/4_2 -lPackageDescription -suppress-warnings -swift-version 4.2 -I /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/usr/lib/swift/pm/4_2 -sdk / /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.dir/tool/Package.swift -fileno 7
02:00:20 warning: no targets to build in package
02:00:20 
02:00:20 $ "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/llvm-linux-x86_64/bin/FileCheck" "--check-prefix" "CHECK-BUILD-LOG" "--input-file" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.build-log" "/home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/swift-integration-tests/test-foundation-package/test-foundation-package.txt"
02:00:20 # command stderr:
02:00:20 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/swift-integration-tests/test-foundation-package/test-foundation-package.txt:15:21: error: expected string not found in input
02:00:20 // CHECK-BUILD-LOG: swiftc{{.*}} -module-name tool
02:00:20                     ^
02:00:20 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.build-log:1:1: note: scanning from here
02:00:20 lsb_release -r
02:00:20 ^
02:00:20 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.build-log:3:764: note: possible intended match here
02:00:20 /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/usr/bin/swiftc --driver-mode=swift -L /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/usr/lib/swift/pm/4_2 -lPackageDescription -suppress-warnings -swift-version 4.2 -I /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/usr/lib/swift/pm/4_2 -sdk / /home/buildnode/jenkins/workspace/oss-swift-package-linux-ubuntu-16_04/build/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/test-foundation-package/Output/test-foundation-package.txt.tmp.dir/tool/Package.swift -fileno 7
02:00:20                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            ^
02:00:20 
02:00:20 error: command failed with exit status: 1
02:00:20 
02:00:20 --
```

**Comment by aciidb0mb3r** [comment]
sorry about that, I’ll fix them first thing in the morning.

**Comment by aciidb0mb3r** [comment]
Disabled the tests in https://github.com/apple/swift-integration-tests/commit/a39ef351c003c74b5a119aa5dab3b6231d045adf

**Pull request merged**